src/components/AudioDetails/Related.vue

```diff
@@ -3,22 +3,59 @@
     <h4 class="b-header mb-6">
       {{ $t('audio-details.related-audios') }}
     </h4>
-    <AudioTrack
-      v-for="audio in relatedAudios"
-      :key="audio.id"
-      :audio="audio"
-      :is-compact="true"
-      layout="row"
-      class="mb-12"
-    />
+    <template v-if="!$fetchState.error">
+      <AudioTrack
+        v-for="audio in audios"
+        :key="audio.id"
+        :audio="audio"
+        layout="row"
+        size="m"
+        class="mb-12"
+      />
+      <LoadingIcon v-show="$fetchState.pending" />
+    </template>
+    <p v-show="!!$fetchState.error">{{ errorMessage }}</p>
   </aside>
 </template>
 
 <script>
+import { ref } from '@nuxtjs/composition-api'
+import { AUDIO } from '~/constants/media'
+import useRelated from '~/composables/use-related'
+
 export default {
   name: 'RelatedAudios',
   props: {
-    relatedAudios: {},
+    audioId: {
+      type: String,
+      required: true,
+    },
+    service: {},
+  },
+  /**
+   * Fetches related audios on `audioId` change
+   * @param props
+   * @param {string} props.audioId
+   * @param {any} props.service
+   * @return {{audios: (Ref<UnwrapRef<[]>>|Ref<AudioDetail[]>)}}
+   */
+  setup(props) {
+    const mainAudioId = ref(props.audioId)
+    const relatedOptions = {
+      mediaType: AUDIO,
+      mediaId: mainAudioId,
+    }
+    // Using service prop to be able to mock when testing
+    if (props.service) {
+      relatedOptions.service = props.service
+    }
+    const { media: audios } = useRelated(relatedOptions)
+    return { audios }
+  },
+  computed: {
+    errorMessage() {
+      return this.$t('media-details.related-error')
+    },
```

Comment:
What's the reason to create a `computed` definition for this rather than inlining the call to `$t` in the template?

Reply:
It was probably from the previous implementation where we added a media type to the message, and the `$t` expression looked really long. But with a more generic message here, it is absolutely unnecessary :)